src/components/molecules/UiLoader/UiLoader.vue

```diff
@@ -89,56 +90,108 @@ const props = defineProps({
     default: 'div',
   },
   /**
-   * Use this props to pas transition name
+   * Use this props to set transition name
    */
-  transition: {
-    type: [String, Boolean] as PropType<string | boolean>,
-    default: false,
+  name: {
+    type: String,
+    default: 'v',
   },
   /**
-   * Use this props to start rendering final component before loading stops, so it's completely rendered
+   * Use this props to apply a transition on the initial render
    */
-  eagerLoadComponent: {
+  appear: {
     type: Boolean,
-    default: false,
+    default: true,
   },
 });
-const component = computed<LoaderComponent>(() => {
+const loaderEl = ref<HTMLElement | null>(null);
+const contentEl = ref<HTMLElement | null>(null);
+const loaderComponent = computed<LoaderComponent>(() => {
   const components: Record<LoaderType, LoaderComponent> = {
     spinner: UiLoaderSpinner,
     ellipsis: UiLoaderEllipsis,
     skeleton: UiLoaderSkeleton,
   };
   return components[props.type];
 });
-const outerComponent = computed<VNode>(() => (
-  props.transition
-    ? h(TransitionGroup, { name: props.transition as string })
-    : h((_, { slots }) => (slots.default as Slot)())
-));
+const animateEl = ref<'loader' | 'content'>(props.isLoading ? 'loader' : 'content');
+const isLoaderAnimate = computed(() => animateEl.value === 'loader');
+const isContentAnimate = computed(() => animateEl.value === 'content');
+const transitionState = reactive({
+  from: false,
+  to: false,
+});
+const loaderClass = computed(() => {
+  const { from, to } = transitionState;
+  const move = props.isLoading ? 'enter' : 'leave';
+  return {
+    [`${props.name}-${move}-from`]: from && isLoaderAnimate.value,
+    [`${props.name}-${move}-active`]: to && isLoaderAnimate.value,
+    [`${props.name}-${move}-to`]: to && isLoaderAnimate.value,
+  };
+});
+const contentClass = computed(() => {
+  const { from, to } = transitionState;
+  const move = !props.isLoading ? 'enter' : 'leave';
+  return {
+    [`${props.name}-${move}-from`]: from && isContentAnimate.value,
+    [`${props.name}-${move}-active`]: to && isContentAnimate.value,
+    [`${props.name}-${move}-to`]: to && isContentAnimate.value,
+  };
+});
+const loaderStyle = computed<CSSProperties>(() => ({
+  visibility: isLoaderAnimate.value ? undefined : 'hidden',
+  position: isLoaderAnimate.value && props.type !== 'ellipsis' ? undefined : 'absolute',
+}));
+const contentStyle = computed<CSSProperties>(() => ({
+  visibility: isContentAnimate.value ? undefined : 'hidden',
+  position: isContentAnimate.value || props.type === 'ellipsis' ? undefined : 'absolute',
+}));
+const initTransition = () => {
+  transitionState.from = true;
+  setTimeout(() => {
+    transitionState.to = true;
+    transitionState.from = false;
+  });
```

Comment:
```suggestion
  nextFrame((() => {
    transitionState.to = true;
    transitionState.from = false;
  });
```
from Vue Transtion component 
```
function nextFrame(cb: () => void) {
  requestAnimationFrame(() => {
    requestAnimationFrame(cb)
  })
}
```

Reply:
Awesome, thanks 🔥